### PR TITLE
 Fix url join with RasaNLUInterpreter

### LINKED ISSUE #3836: RasaNLUHttpInterpreter makes call to wrong URL for parsing intent
**Rasa version**: 1.1.3

**Rasa X version** (if used & relevant):

**Python version**: 3.6

**Operating system** (windows, osx, ...): OSX

**Issue**:

RASA allows a custom URL for NLU endpoint, by specifying the NLU endpoint in endpoints.yml. But currently, at the time of making HTTP call to the NLU endpoint, the URL constructed is not right.
If one aims to set up an NLU endpoint at ``` http://localhost:8080/a/b/c/model/parse/ ```, so they will have the following endpoint configuration for NLU endpoint.

```
nlu:
  url: 'http://localhost:8080/a/b/c/'
```
But the issue happens in the code while doing URL join for 
```urljoin(self.endpoint.url, "/model/parse")```

Sample Code Issue Snippet:
For Python 3.6 and 3.7 as well

```
>>> urljoin('http://localhost:8080/a/b/c/','/model/parse')
'http://localhost:8080/model/parse'
>>> # Clearly, we have lost a part from URL, which is a bug
...
>>>
>>>
>>> urljoin('http://localhost:8080/a/b/c/','model/parse')
'http://localhost:8080/a/b/c/model/parse'
>>> # Removing forward slash from the second argument, fixes the issue
```

Code Issue in Rasa: https://github.com/RasaHQ/rasa/blob/master/rasa/core/interpreter.py#L212

Proposal:

Change from:
```urljoin(self.endpoint.url, "/model/parse") ```
to
```urljoin(self.endpoint.url, "model/parse") ```

### PR DESCRIPTION
**Pull request by RanaMostafaAbdElMohsen**
Fixes  #3836
**Proposed changes**:
- Fix urljoin in RasaNLUInterpreter calling right url 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by psds01** [comment]
I don't think it passes all the tests required. Check [this](https://github.com/RasaHQ/rasa/issues/3836#issuecomment-511358731) comment for tests and [this](https://github.com/RasaHQ/rasa/issues/3836#issuecomment-511358947) for solution.

Tried few examples, found out that for a base url of form `http://localhost:8080/a/b/c`  or `http://localhost:8080/a/b/c/`, model_path of the form `/e/f/` won't work, where we are trying to do urljoin(base_url, model_path)

**Comment by RanaMostafaAbdElMohsen** [comment]
@akelad ,
Please check this pull request.

**Comment by akelad** [comment]
Thanks for submitting this PR, we'll give it a review as soon as possible.

File: tests/core/test_interpreter.py [hunk 1]
```diff
@@ -103,3 +103,35 @@ def regex_interpreter():
         response = {"text": "message_text", "token": None, "message_id": "message_id"}
 
         assert query == response
+
+
+async def test_http_interpreter_no_slash_at_end():
```
Comment (tabergma) [inline tests/core/test_interpreter.py]: You can use parametrize to reduce code duplication and define only one test (https://docs.pytest.org/en/latest/parametrize.html). We use it quite often in our test. Here is an example: https://github.com/RasaHQ/rasa/blob/9d1afe0c1c06c9c787a30f99e3deb0bc5bf01620/tests/test_utils.py#L75

File: rasa/core/interpreter.py [hunk 2]
```diff
@@ -211,7 +211,10 @@ def __init__(self, endpoint: EndpointConfig = None) -> None:
 
         params = {"token": self.endpoint.token, "text": text, "message_id": message_id}
 
-        url = urljoin(self.endpoint.url, "/model/parse")
+        if self.endpoint.url.endswith("/"):
```
Comment (tabergma) [inline rasa/core/interpreter.py]: Why no using the same logic for both cases? 
So either
```
if not self.endpoint.url.endswith("/"):
    self.endpoint.url = self.endpoint.url + "/"

url = urljoin(self.endpoint.url, "model/parse")
```
or
```
if self.endpoint.url.endswith("/"):
    url = self.endpoint.url + "model/parse"
else:
    url = self.endpoint.url + "/model/parse"  
```

Comment (RanaMostafaAbdElMohsen) [reply rasa/core/interpreter.py]: Thanks. Resolved

File: tests/core/test_interpreter.py (hunk 1 above)
Comment (RanaMostafaAbdElMohsen) [reply tests/core/test_interpreter.py]: Thanks. Resolved

**Pull request merged**